contracts/HATKlerosConnector.sol

```diff
@@ -0,0 +1,711 @@
+// SPDX-License-Identifier: MIT
+
+/**
+ *  @authors: [@unknownunknown1]
+ *  @reviewers: []
+ *  @auditors: []
+ *  @bounties: []
+ *  @deployments: []
+ */
+
+pragma solidity 0.8.16;
+
+import {IDisputeResolver, IArbitrator} from "@kleros/dispute-resolver-interface-contract/contracts/IDisputeResolver.sol";
+import "./HATVault.sol";
+import "./libraries/CappedMath.sol";
+
+/**
+ *  @title HATKlerosConnector
+ *  @dev This contract acts a connector between HatsFinance and Kleros court.
+ *  This contract trusts that the Arbitrator is honest and will not reenter or modify its costs during a call.
+ *  The arbitrator must support appeal period.
+ *  The contract also trusts that HATVault contract is honest and won't reenter.
+ */
+contract HATKlerosConnector is IDisputeResolver {
+    using CappedMath for uint256;
+
+    uint256 private constant RULING_OPTIONS = 2; // The amount of non 0 choices the arbitrator can give.
+    uint256 private constant MULTIPLIER_DIVISOR = 10000; // Divisor parameter for multipliers.
+
+    enum Status {
+        None, // Claim is open to challenge.
+        DisputedByClaimant, // Claim was challenged by the hacker.
+        DisputedByDepositor, // Claim was challenged by the pool participant.
+        Resolved // Claim was resolved by the arbitrator.
+    }
+    
+    enum Decision {
+        None, // Court wasn't able to make a decisive ruling. In this case the claim is unchanged. Both sides will get their appeal deposits back in this case.
+        MakeNoChanges, // Accept the claim as it is without changing.
+        SideWithChallenger // Side with the challenger and either change claim's severity, dismiss it altogether or submit a new claim, depending on who created a dispute.
+    }
+
+    struct Claim {
+        Status status; // Claim's current status.
+        uint16 bountyPercentage; // Bounty that the claimant demands.
```

Comment:
we'd also like to add an `address beneficiary` option here (i.e. the challenger can also propose to change the beneficiary)